tests/js/client/shell/shell-transaction-intermediate-commit-babies-cluster.js

```diff
@@ -0,0 +1,306 @@
+      let shards = db._collection(cn).shards(true);
+      let shardId = Object.keys(shards)[0];
+      let leader = getEndpointById(shards[shardId][0]);
+
+      let docs = [];
+      for (let i = 1; i <= 1000; ++i) {
+        docs.push({ _key: "test" + i });
+      }
+
+      let intermediateCommitsBefore = getMetric(leader, "arangodb_intermediate_commits_total");
+
+      c.insert(docs);
+
+      // no intermediate commits
+      let intermediateCommitsAfter = getMetric(leader, "arangodb_intermediate_commits_total");
+      assertEqual(intermediateCommitsBefore, intermediateCommitsAfter);
+
+      assertEqual(1000, db._collection(cn).count());
+    },
+      
+    testDocumentsSingleArrayIntermediateCommitsSuppressed: function () {
+      let c = db._create(cn, { numberOfShards: 1, replicationFactor: 2 });
+      let shards = c.shards(true);
+      let shardId = Object.keys(shards)[0];
+      let leader = getEndpointById(shards[shardId][0]);
+      // trigger a certain failure point on the leader
+      debugSetFailAt(leader, "TransactionState::intermediateCommitCount100");
+      
+      let docs = [];
+      for (let i = 1; i <= 1000; ++i) {
+        docs.push({ _key: "test" + i });
+      }
+      
+      let intermediateCommitsBefore = getMetric(leader, "arangodb_intermediate_commits_total");
+
+      c.insert(docs);
+
+      // no intermediate commit must have happened
+      let intermediateCommitsAfter = getMetric(leader, "arangodb_intermediate_commits_total");
+      assertEqual(intermediateCommitsBefore, intermediateCommitsAfter);
+
+      assertEqual(1000, c.count());
+    },
+    
+    testDocumentsTransaction: function () {
+      let c = db._create(cn, { numberOfShards: 1, replicationFactor: 1 });
+      let shards = db._collection(cn).shards(true);
+      let shardId = Object.keys(shards)[0];
+      let leader = getEndpointById(shards[shardId][0]);
+
+      let docs = [];
+      for (let i = 1; i <= 1000; ++i) {
+        docs.push({ _key: "test" + i });
+      }
+        
+      let intermediateCommitsBefore = getMetric(leader, "arangodb_intermediate_commits_total");
+
+      let opts = { collections: { write: cn } }; 
+      let trx = db._createTransaction(opts);
+
+      try {
+        trx.collection(cn).insert(docs);
+
+        trx.commit();
+      } catch (err) {
+        trx.abort();
+      }
+
+      // no intermediate commits
+      let intermediateCommitsAfter = getMetric(leader, "arangodb_intermediate_commits_total");
+      assertEqual(intermediateCommitsBefore, intermediateCommitsAfter);
+        
+      assertEqual(1000, db._collection(cn).count());
+    },
+    
+    testDocumentsTransactionIntermediateCommitsSuppressed: function () {
+      let c = db._create(cn, { numberOfShards: 1, replicationFactor: 1 });
+      let shards = db._collection(cn).shards(true);
+      let shardId = Object.keys(shards)[0];
+      let leader = getEndpointById(shards[shardId][0]);
```

Comment:
This initialization code is almost identical for all testcases, though some use rf=2 and some rf=1. I would prefer if we could split these tests into two separate testsuites and move the common initialization code into the respective setup functions.

Reply:
Actually the setups are slightly different. I thus created 2 separate testsuites out of it so it becomes a bit clearer.